daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/FrontStack.scala

```diff
@@ -129,6 +144,17 @@ object FrontStack {
 
   def unapply[T](xs: FrontStack[T]): Boolean = xs.isEmpty
 
+  private[data] final class FSCanBuildFrom[A]
```

Comment:
Curiosity: do you know what's going to happen to `CanBuildFrom` in the upcoming Scala versions? I believe it's on its way out starting from 2.13, right?

Reply:
It still has its use cases and accordingly replacements in 2.13, which we would define, in fact they are even high-_er_ kinded; the idea of doing away with the typeclass entirely was never in the cards, practically speaking.

Reply:
Since most of your other suggestions are CanBuildFrom related, I'll expand:

CanBuildFrom was used so ubiquitously to try to avoid having higher-kinded types permeate the library. So it was only natural that when they tried to reduce (not completely eliminate, as I already stated) its usage in 2.13, they ended up with...[ubiquitous higher-kinded types](https://github.com/scala/scala/blob/v2.13.0/src/library/scala/collection/immutable/List.scala#L85-L88). So, many cases that used to use CBF as an implicit-based method to do HK type application without HKs, instead just throw up their hands and directly use HKs. An outcome I'm perfectly satisfied with.

So when it comes to those operations that will be HK'd in 2.13, like map, collect, ++, I'm thinking about phasing out breakOut in new code. Not really having a rule against it, because it's still just too useful, but more thinking about making the 2.13 transition easier on us. For example, if we want to fuse some operations, it might be good to try putting them on `iterator` or `view`, terminating with `to`, all of which exist in 2.12 and 2.13 (though the pattern of usage is a little different for `to`).

2.13 collections have a collection (sorry) of new type...classes to play some different roles of `CanBuildFrom`, like [the `*Factory` types](https://github.com/scala/scala/blob/v2.13.0/src/library/scala/collection/Factory.scala), and, er, [BuildFrom](https://github.com/scala/scala/blob/v2.13.0/src/library/scala/collection/BuildFrom.scala). A rose by any other name is a completely different type, you know. Among these are straightforward replacements for everything we currently define, so I'm not worried about the 2.13 transition for daml.

Reply:
Great, thanks!